typespec-tests/src/test/java/com/cadl/union/UnionTests.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.cadl.union;
+
+import com.azure.core.util.BinaryData;
+import com.cadl.union.models.Result;
+import com.cadl.union.models.SubResult;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+
+public class UnionTests {
+
+    private static final byte[] bytes = new byte[] { 'a' };
+
+    @Test
+    public void testSerialization() {
+        // test non-null Union (required property)
+        Result result = new Result("name", BinaryData.fromObject(bytes));
+        BinaryData json = BinaryData.fromObject(result);
```

Comment:
nit: Putting the test here is a bit strange to me as we are testing the `BinaryData` serialization/deserialization logic, seems it has nothing to do with tsp. But seems we don't have test suites to test the stream-style serialization's serialization and desefialization cases.. So I'm OK with it for now.

Reply:
BinaryData is just a wrapper for the serialization and de-serialization of JSON <-> class.